src/main/kotlin/org/rust/ide/utils/CallInfo.kt

```diff
@@ -38,11 +65,25 @@ class CallInfo private constructor(
         }
 
         fun resolve(methodCall: RsMethodCall): CallInfo? {
-            return (methodCall.reference.resolve() as? RsFunction)?.let { CallInfo(it) }
+            val function = (methodCall.reference.resolve() as? RsFunction) ?: return null
+            val substitution = let {
```

Comment:
Probably I'm missing something, could you please explain why we cannot use something like `(methodCall.parentDotExpr.type as? TyFunction)?.paramTypes` as the actual types instead of these substitutions?

Reply:
`methodCall.parentDotExpr.type` is the type of the return value of the method call, not of the method (function) itself. I haven't found any easier way how to get the actual substituted method type with generics resolved.